src/webui/api/searchcontroller.cpp

```diff
@@ -0,0 +1,286 @@
+
+void SearchController::resultsAction()
+{
+    checkParams({"id"});
+
+    const QString id = params()["id"].trimmed();
+    if (!m_searches.contains(id)) {
+        setResult(QJsonObject::fromVariantMap(getResults(QList<SearchResult>(), false, 0)));
+        return;
+    }
+
+    WebSearchHandler *webSearchHandler = m_searches[id];
+    const QList<SearchResult> searchResults = webSearchHandler->getSearchResults();
+
+    setResult(QJsonObject::fromVariantMap(getResults(searchResults, webSearchHandler->isActive(), webSearchHandler->getQueueSize())));
+
+    if (!webSearchHandler->isActive())
+        m_searches.remove(id);
+}
+
+void SearchController::downloadTorrent()
+{
+    checkParams({"id", "torrentUrl", "siteUrl"});
+
+    const QString id = params()["id"].trimmed();
+    const QString torrentUrl = params()["torrentUrl"].trimmed();
+    const QString siteUrl = params()["siteUrl"].trimmed();
+
+    WebSearchHandler *webSearchHandler = m_searches[id];
+    if (!m_searches.contains(id)) {
+        throw APIError(APIErrorType::BadParams);
+    }
+
+    webSearchHandler->downloadTorrent(torrentUrl, siteUrl);
+}
+
+void SearchController::categoriesAction()
+{
+    QStringList categories;
+    const QString name = params()["name"].trimmed();
+
+    categories << SearchPluginManager::categoryFullName("all");
+    foreach (const QString &category, SearchPluginManager::instance()->getPluginCategories(name))
+        categories << SearchPluginManager::categoryFullName(category);
+
+    setResult(categories.join(","));
+}
+
+void SearchController::pluginsAction()
+{
+    const QStringList allPlugins = SearchPluginManager::instance()->allPlugins();
+    setResult(QJsonArray::fromVariantList(getPluginsInfo(allPlugins)));
+}
+
+void SearchController::installPluginAction()
+{
+    checkParams({"sources"});
+
+    const QStringList sources = params()["sources"].split('|');
+    for (const QString &source : sources)
+        SearchPluginManager::instance()->installPlugin(source);
+}
+
+void SearchController::uninstallPluginAction()
+{
+    checkParams({"names"});
+
+    const QStringList names = params()["names"].split('|');
+    for (const QString &name : names)
+        SearchPluginManager::instance()->uninstallPlugin(name.trimmed());
+}
+
+void SearchController::enablePluginAction()
+{
+    checkParams({"names", "enable"});
+
+    const QStringList names = params()["names"].split('|');
+    const bool enable = parseBool(params()["enable"].trimmed(), false);
+
+    for (const QString &name : names)
+        SearchPluginManager::instance()->enablePlugin(name.trimmed(), enable);
+}
+
+void SearchController::updatePluginsAction()
+{
+    SearchPluginManager *pluginManager = SearchPluginManager::instance();
+
+    connect(pluginManager, &SearchPluginManager::checkForUpdatesFinished, this, &SearchController::checkForUpdatesFinished);
+    // connect(pluginManager, &SearchPluginManager::checkForUpdatesFailed, this, &SearchController::checkForUpdatesFailed);
```

Comment:
Can you explain your plan? or just remove it.

Reply:
I haven't thought of a good way of handling this. Removing.